### PR TITLE
Protect baseline readers from disposal while in use

### PR DESCRIPTION
**Pull request by tmat**
Will help diagnose errors like [AB#1363929](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1363929)

### CODE REVIEW

**Comment by tmat** [comment]
@davidwengier

**Review by davidwengier — APPROVED** [review]

**Review by ryzngard — APPROVED** [review]

File: src/Features/Core/Portable/EditAndContinue/DebuggingSession.cs [hunk 1]
```diff
@@ -49,12 +49,20 @@ internal sealed class DebuggingSession : IDisposable
...
         private readonly List<IDisposable> _initialBaselineModuleReaders = new();
         private readonly object _projectEmitBaselinesGuard = new();
 
+        /// <summary>
+        /// To avoid accessing metadata/symbol readers that have been disposed,
+        /// read lock is acquired before every operation that may access a baseline module/symbol reader 
+        /// and write lock when the baseline readers are being disposed.
+        /// </summary>
+        private readonly ReaderWriterLockSlim _baselineAccessLock = new();
+        private bool _isDisposed;
```
Comment (sharwell) [inline src/Features/Core/Portable/EditAndContinue/DebuggingSession.cs:64]: ❓ Why not use `ReferenceCountedDisposable<T>` to guarantee concurrently-used resources are not disposed while still in use?

**Review by sharwell — DISMISSED** [review]
One question

File: src/Features/Core/Portable/EditAndContinue/DebuggingSession.cs (hunk 1 above)
Comment (tmat) [reply src/Features/Core/Portable/EditAndContinue/DebuggingSession.cs:64]: I don't see how ReferenceCountedDisposable is applicable here.

**Pull request merged**